### PR TITLE
Adds the first and last name provider for names in Bosnia and Herzegovina

### PR DESCRIPTION
**Pull request by elahmo**
Adds the first and last name provider for names from Bosnian language.

### CODE REVIEW

**Comment by fcurella** [comment]
Thank you! Can you also add a testcase for the new provider?

File: faker/providers/person/bs_BA/__init__.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        "Sara", "Sarah", "Šejla", "Selma", "Sumeja", "Tajra", "Uma", "Una",
+        "Zana", "Zara", "Zejneb", "Zerina",
+    ]
+
+    first_names = first_names_female + first_names_male
+    """
+    Last names were taken from Wikipedia Category listing about bosnian surnames
+    https://en.wikipedia.org/wiki/Category:Bosnian-language_surnames
+    """
+    last_names = ["Abadžić", "Ademović", "Adilović", "Aganović", "Ahmetović",
```
Comment (fcurella) [inline faker/providers/person/bs_BA/__init__.py]: add a newline after the `[`:

```python
last_names = [
    "Abadžić",
    # ...
]
```

File: faker/providers/person/bs_BA/__init__.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
+# coding=utf-8
+from __future__ import unicode_literals
+from .. import Provider as PersonProvider
+
+
+class Provider(PersonProvider):
+    """
+    The names used here are the top 100 names released by the Federal statistics
```
Comment (fcurella) [inline faker/providers/person/bs_BA/__init__.py]: there are some trailing spaces at the end of hits and the next line. Can you remove them?

**Comment by elahmo** [comment]
> Thank you! Can you also add a testcase for the new provider?

Sure! Are there any guidelines what should be tested?

**Comment by fcurella** [comment]
@elahmo You can look at https://github.com/joke2k/faker/blob/master/tests/providers/test_person.py#L162-L181 for an example

File: faker/providers/person/bs_BA/__init__.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        "Mustafić", "Mutapčić", "Muzaferi", "Muzurović", "Nadarević", "Nikić",
+        "Nišić", "Pačar", "Pašalić", "Pjanić", "Plakalović", "Predojević",
+        "Preldžić", "Rahimić", "Rakić", "Rastod", "Sadiković", "Salatić",
+        "Salihović", "Salkić", "Saračević", "Sarajlić", "Šarić", "Šećerović",
+        "Seferović", "Šehić", "Selimović", "Šetkić", "Sidr", "Sijerčić",
+        "Šimunović", "Šišić", "Skenderović", "Smajić", "Smajlović", "Sokolović",
+        "Suljić", "Sušić", "Tahirović", "Tanković", "Terzić", "Tinjić",
+        "Topalović", "Topčagić", "Toskić", "Turković", "Varešanović", "Velagić",
+        "Vidić", "Vučkić", "Vugdalić", "Vuković", "Zahirović", "Zakarić",
+        "Zukanović",
```
Comment (fcurella) [inline faker/providers/person/bs_BA/__init__.py]: there is a trailing space here. could you remove it?

**Pull request closed without merging**